### PR TITLE
New case: complete a standby commit job

### PR DESCRIPTION
**Pull request by aliang123**
blockdev_commit_standby:complete a standby commit job

Signed-off-by: Aihua Liang <aliang@redhat.com>

id: 1975687

### CODE REVIEW

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_commit_standby.q35: PASS (96.21 s)

**Comment by aliang123** [comment]
@ngu-niny, @zhencliu, please help to review, thanks

**Comment by zhencliu** [comment]
LGTM

File: qemu/tests/blockdev_commit_standby.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        cmd, args = commit_cmd(device)
+        self.main_vm.monitor.cmd(cmd, args)
+        job_id = args.get("job-id", device)
+        job_utils.wait_until_job_status_match(self.main_vm, 'ready', job_id, timeout=120)
+        self.main_vm.monitor.cmd("job-pause", {'id': job_id})
+        if not job_utils.is_block_job_paused(self.main_vm, job_id):
+            self.test.fail("Block job is not paused")
+        self.main_vm.monitor.cmd("job-complete", {'id': job_id})
+        self.main_vm.monitor.cmd("job-resume", {'id': job_id})
+        job_utils.wait_until_block_job_completed(self.main_vm, job_id)
```
Comment (ngu-niny) [inline qemu/tests/blockdev_commit_standby.py:24]: @aliang123 The 2nd time 'job-complete' would be executed in the function. It seems a redundancy.

Comment (aliang123) [reply qemu/tests/blockdev_commit_standby.py:23]: @ngu-niny, have updated, please help to check again, thanks.

Comment (ngu-niny) [reply qemu/tests/blockdev_commit_standby.py:23]: @aliang123 It's better to move the "job-complete" command to job_utils.wait_until_block_job_completed() function too.

Comment (aliang123) [reply qemu/tests/blockdev_commit_standby.py:22]: @ngu-niny, have updated. Please help to check again.

**Comment by ngu-niny** [comment]
LGTM.

 (1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.blockdev_commit_standby:  PASS (156.82 s)

Acked-by: Nini Gu <ngu@redhat.com>

File: qemu/tests/blockdev_commit_standby.py (hunk 1 above)
Comment (zhencliu) [reply qemu/tests/blockdev_commit_standby.py:22]: I am afraid we cannot move all operations into the wait_until_block_job_completed function, which is widely used, the change might cause regressions. what about:

self.main_vm.monitor.cmd("job-pause", {'id': job_id})
job_utils.block_job_complete()
self.main_vm.monitor.cmd("job-resume", {'id': job_id})
if job_utils.get_event_by_condition("BLOCK_JOB_COMPLETED") is None:
case fail.

This explicitly follow your polarion test steps, and no code change in job_utils.py
Finally, if we wanna check the job is removed, we can use
job_utils.wait_until_job_status_match("null")

Comment (aliang123) [reply qemu/tests/blockdev_commit_standby.py:22]: zhenchao, after self.main_vm.monitor.cmd("job-pause", {'id': job_id}), job status reach 'standby', but in job_utils.block_job_complete(), we need to match job status 'ready'. So vm.monitor.cmd("job-complete", {'id': job_id}) needed here.

And "job_utils.get_event_by_condition("BLOCK_JOB_COMPLETED") is None" is a good solution, I think.

Comment (zhencliu) [reply qemu/tests/blockdev_commit_standby.py:22]: OK, right, let's send the job-complete command directly.

**Comment by aliang123** [comment]
(1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.x86_64.io-github-autotest-qemu.blockdev_commit_standby.q35: PASS

**Comment by aliang123** [comment]
@zhencliu, @ngu-niny  please help to review again, thanks.

File: qemu/tests/blockdev_commit_standby.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        cmd, args = commit_cmd(device)
+        self.main_vm.monitor.cmd(cmd, args)
+        job_id = args.get("job-id", device)
+        job_utils.wait_until_job_status_match(self.main_vm, "ready", job_id, timeout=120)
+        self.main_vm.monitor.cmd("job-pause", {"id": job_id})
+        if not job_utils.is_block_job_paused(self.main_vm, job_id):
+            self.test.fail("Block job is not paused")
+        self.main_vm.monitor.cmd("job-complete", {"id": job_id})
+        self.main_vm.monitor.cmd("job-resume", {"id": job_id})
+        if not job_utils.get_event_by_condition(self.main_vm, "BLOCK_JOB_COMPLETED"):
```
Comment (zhencliu) [inline qemu/tests/blockdev_commit_standby.py:24]: It's better to use the job_id to filter the job, but here, it's OK for there is only one.

**Comment by zhencliu** [comment]
LGTM

**Comment by ngu-niny** [comment]
LGTM

 (1/1) Host_RHEL.m8.u5.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.5.0.ppc64le.io-github-autotest-qemu.blockdev_commit_standby:  PASS (176.46 s)

**Pull request merged**